packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx

```diff
@@ -44,20 +44,31 @@ type FilterProps = {
 };
 
 const VisitTable: React.FC<VisitTableProps> = ({ showAllEncounters, visits }) => {
-  const encountersCount = 20;
+  const visitCount = 20;
   const { t } = useTranslation();
-  const encounterTypes = [...new Set(visits.map((encounter) => encounter.encounterType))].sort();
-  const { results: paginatedEncounters, goTo, currentPage } = usePagination(visits ?? [], encountersCount);
   const isTablet = useLayoutType() === 'tablet';
-  const [filteredRows, setFilteredRows] = useState<Array<MappedEncounter>>([]);
+
+  const encounterTypes = [...new Set(visits.map((encounter) => encounter.encounterType))].sort();
+
   const [filter, setFilter] = useState('');
 
-  useEffect(() => {
+  const filteredRows = useMemo(() => {
+    if (!filter || filter == 'All') {
+      return visits;
+    }
+
     if (filter) {
-      setFilteredRows(visits?.filter((encounter) => encounter.encounterType === filter));
-      setFilter('');
+      return visits?.filter((encounter) => encounter.encounterType === filter);
     }
-  }, [filter, filteredRows, visits]);
+
+    return visits;
+  }, [filter, visits]);
+
+  const {
+    results: paginatedVisits,
+    goTo,
+    currentPage,
+  } = usePagination(filteredRows ? filteredRows : visits ?? [], visitCount);
```

Comment:
Can't you just use `filteredRows` since they are returning `visits` there is no filter.

Reply:
Sweet, thanks.